docs/quickstart/java_api_quickstart.md

```diff
@@ -108,16 +118,201 @@ In Eclipse, choose
+    myShadowJar.exclude group: 'com.google.code.findbugs', module: 'jsr305'
+    myShadowJar.exclude group: 'org.slf4j'
+    myShadowJar.exclude group: 'log4j'
+}
+
+// declare the dependencies for your production and test code
+dependencies {
+    compile "org.apache.flink:flink-java:${flinkVersion}"
+    compile "org.apache.flink:flink-streaming-java_${scalaBinaryVersion}:${flinkVersion}"
+
+    //myShadowJar "org.apache.flink:flink-connector-kafka-0.11_${scalaBinaryVersion}:${flinkVersion}"
+
+    compile "log4j:log4j:${log4jVersion}"
+    compile "org.slf4j:slf4j-log4j12:${slf4jVersion}"
+
+    testCompile "junit:junit:4.12"
+}
+
+// make compileOnly dependencies available for tests:
+sourceSets {
+    main.runtimeClasspath += configurations.myShadowJar
+    main.runtimeClasspath += configurations.myShadowJar
+
+    test.compileClasspath += configurations.myShadowJar
+    test.runtimeClasspath += configurations.myShadowJar
+}
+
+jar {
+    manifest {
+        attributes 'Built-By': System.getProperty('user.name'),
+                'Build-Jdk': System.getProperty('java.version')
+    }
+}
+
+shadowJar {
+    configurations = [project.configurations.myShadowJar]
+}
+                {% endhighlight %}
+            </div>
+            <div class="tab-pane" id="gradle-settings">
+                {% highlight gradle %}
+rootProject.name = 'quickstart'
+                {% endhighlight %}
+            </div>
+        </div>
+    </div>
+
+    <div class="tab-pane" id="gradle-script">
+    {% highlight bash %}
+    $ curl https://flink.apache.org/q/gradle-quickstart.sh | bash
+    {% endhighlight %}
+    This allows you to <strong>name your newly created project</strong>. It will interactively ask
+    you for the project name, organization (also used for the package name), project version,
+    Scala and Flink version.
+    </div>
+</div>
+
+### Inspect Project
+
+There will be a new directory in your working directory based on the
+project name you provided, e.g. for `quickstart`:
+
+{% highlight bash %}
+$ tree quickstart/
+quickstart/
+├── README
+├── build.gradle
+├── settings.gradle
+└── src
+    └── main
+        ├── java
+        │   └── org
+        │       └── myorg
+        │           └── quickstart
+        │               ├── BatchJob.java
+        │               └── StreamingJob.java
+        └── resources
+            └── log4j.properties
+{% endhighlight %}
+
+The sample project is a __Gradle project__, which contains two classes: _StreamingJob_ and _BatchJob_ are the basic skeleton programs for a *DataStream* and *DataSet* program.
+The _main_ method is the entry point of the program, both for in-IDE testing/execution and for proper deployments.
+
+We recommend you __import this project into your IDE__ to develop and
+test it. IntelliJ IDEA supports Gradle projects after installing the `Gradle` plugin.
+Eclipse does so via the [Eclipse Buildship](https://projects.eclipse.org/projects/tools.buildship) plugin.
```

Comment:
no, sorry - that should of course be double-checked

Reply:
just checked - seems to be working (you need to manually select a gradle version >= 3.0 because the default gradle wrapper uses 2.14 which is too old) - I added appropriate comments into the fixup commit